js-intro.js

```diff
@@ -7,59 +7,98 @@ var mantra = "Be the dev"
 
 // 1a. Write the code that determines if there is a 'B' in mantra.
 
-
-
+const lettersIn = (str) => {
+  return str.includes('B') ? str + ' has the letter B': str + ' does not have the letter B'
+}
+console.log(lettersIn(mantra));
 // 1b. Write the code that determines if there is an 'x' in mantra.
 
+const lettersIn1 = (str) => {
+  return str.includes('x') ? str + ' has the letter x': str + ' does not have the letter x'
+}
+console.log(lettersIn1(mantra));
 
 
 // 1c. Write the code that determines if there is a 'v' in mantra.
-
-
+const lettersIn2 = (str) => {
+  return str.includes('v') ? str + ' has the letter v': str + ' does not have the letter v'
+}
+console.log(lettersIn2(mantra));
 
 // 1d. STRETCH: Write a function that can take two arguments - a string and a letter and returns a sentence telling the user if the letter exists within the string.
 
-
-
+const letterInStr = (letter, str) => {
+  return str.includes(letter) ? `${letter} is in ${str}`: `${letter} is not ${str}`
+}
+console.log(letterInStr('d', mantra));
 // ------------------------------ Consider the following variables:
+// ---------------------------------------------------------------------------//
 
 var myDog = "Snoopy"
 var myCat = "Garfield"
 
 // 2a. Create a function that returns the longer of the two variables. Expected output: "Garfield"
 
-
-
+const longerWord = (str1, str2) => {
+    if (str1.length > str2.length) {
+        return str1; 
+    } else {
+        return str2; 
+    }
+}
+console.log(longerWord(myDog, myCat));
+console.log(longerWord(myCat, myDog));
 
 // 2b. Create a function that takes in two arguments and returns an array containing the two variables. Expected output: ["Snoopy", "Garfield"]
 
-
-
+const joinStr = (str1, str2) => {
+  let strArr = []; 
+    strArr.push(str1, str2)
+    return strArr
+}
+console.log(joinStr(myDog, myCat))
 
 // 2c. STRETCH: Write a function that returns the two variables in all lowercase letters. Expected output: "snoopy garfield"
+const lowerCase = (str1, str2) => {
+  console.log(str1.concat(' ', str2).toLowerCase());
+}
 
-
-
-
+lowerCase(myDog, myCat)
 // ------------------------------ Consider the following variable:
 
 var myMessage = "Learning is fun!"
 
 // 3a. Write the code that logs each letter of the message using a for loop.
-
-
+for (var i = 0; i < myMessage.length; i++) {
+  console.log(myMessage[i])
+}
 
 // 3b. Write the code that logs each letter of the message using map.
 
-
+const sepLetters = (arr) => {
+  let betterArr =[]
+  for (var i = 0; i < arr.length; i++) {
+    betterArr.push(arr[i])
+  }
+  return betterArr;
+}
+console.log(myMessage.split('').map(sepLetters))
 
```

Comment:
Map includes the ability to iterate over the array. You won't need the for loop if you are using map.